source/uwp/Renderer/lib/AdaptiveRenderContext.cpp

```diff
@@ -3,254 +3,124 @@
     {
-        return m_textStyle.CopyTo(textStyle);
+        if (auto renderResult = GetRenderResult())
+        {
+            renderResult->AddInputValue(inputValue, renderArgs);
+        }
     }
 
-    HRESULT AdaptiveRenderContext::put_TextStyle(_In_ ABI::Windows::Foundation::IReference<ABI::AdaptiveCards::ObjectModel::Uwp::TextStyle>* textStyle)
+    void AdaptiveRenderContext::LinkSubmitActionToCard(ObjectModel::Uwp::IAdaptiveActionElement const& submitAction,
+                                                       Uwp::AdaptiveRenderArgs const& renderArgs)
     {
-        m_textStyle = textStyle;
-        return S_OK;
+        if (auto renderResult = GetRenderResult())
+        {
+            renderResult->LinkActionToCard(submitAction, renderArgs);
+        }
     }
 
-    HRESULT AdaptiveRenderContext::get_HorizontalContentAlignment(
-        _Outptr_ ABI::Windows::Foundation::IReference<ABI::AdaptiveCards::ObjectModel::Uwp::HAlignment>** horizontalAlignment)
+    void AdaptiveRenderContext::LinkCardToParent(ObjectModel::Uwp::AdaptiveCard const& card, Uwp::AdaptiveRenderArgs const& args)
     {
-        return m_horizontalAlignment.CopyTo(horizontalAlignment);
+        if (auto renderResult = GetRenderResult())
+        {
+            renderResult->LinkCardToParent(card, args);
+        }
     }
 
-    HRESULT AdaptiveRenderContext::put_HorizontalContentAlignment(
-        _In_ ABI::Windows::Foundation::IReference<ABI::AdaptiveCards::ObjectModel::Uwp::HAlignment>* horizontalAlignment)
+    Uwp::IAdaptiveInputValue AdaptiveRenderContext::GetInputValue(ObjectModel::Uwp::IAdaptiveInputElement const& inputElement)
     {
-        m_horizontalAlignment = horizontalAlignment;
-        return S_OK;
+        if (auto renderResult = GetRenderResult())
+        {
+            return renderResult->GetInputValue(inputElement);
+        }
+        else
+        {
+            return nullptr;
+        }
     }
 
-    Microsoft::WRL::ComPtr<ABI::Windows::UI::Xaml::IResourceDictionary> AdaptiveRenderContext::GetDefaultActionSentimentDictionary()
+    winrt::Windows::UI::Xaml::ResourceDictionary AdaptiveRenderContext::GetDefaultActionSentimentDictionary()
     {
         return m_actionSentimentDefaultDictionary;
     }
 
-    HRESULT AdaptiveRenderContext::GetRenderResult(AdaptiveCards::Rendering::Uwp::RenderedAdaptiveCard** renderResult)
+    winrt::com_ptr<implementation::RenderedAdaptiveCard> AdaptiveRenderContext::GetRenderResult()
     {
-        ComPtr<IRenderedAdaptiveCard> strongRenderResult;
-        RETURN_IF_FAILED(m_weakRenderResult.As(&strongRenderResult));
-        RETURN_IF_FAILED(strongRenderResult == nullptr ? E_FAIL : S_OK);
-        ComPtr<RenderedAdaptiveCard> renderResultObject = PeekInnards<RenderedAdaptiveCard>(strongRenderResult);
-        RETURN_IF_FAILED(renderResultObject == nullptr ? E_FAIL : S_OK);
```

Comment:
I've added:
`throw(winrt::hresult_error(winrt::hresult_invalid_argument(
            L"RenderResult could not be resolved. Most likely, the object is no longer alive because all the references were destroyed.")));`

Reply:
@paulcam206